ibis/expr/types.py

```diff
@@ -1016,6 +1016,31 @@ def _implicit_cast(self, arg):
         return DateScalar(op)
 
 
+class TimeValue(TemporalValue):
+
+    def type(self):
+        return dt.time
+
+    def _can_implicit_cast(self, arg):
+        op = arg.op()
+        if isinstance(op, Literal):
+            try:
+                from pandas.core.tools.datetimes import to_time
+                to_time(op.value)
+                return True
+            except ValueError:
+                return False
+        return False
+
+    def _can_compare(self, other):
+        return isinstance(other, (TemporalValue, StringValue))
```

Comment:
So I don't think `TemporalValue` can be here, just `TimeValue` and `StringValue`. It makes sense to compare a `TimestampValue` with a `DateValue` and vice versa, but not a `TimestampValue` with `TimeValue` (because the time value has no date so there's no ordering) or `DateValue` with `TimeValue` for the same reason.